WordPress/Classes/ViewRelated/Post/AuthorFilterButton.swift

```diff
@@ -0,0 +1,120 @@
+        imageView.contentMode = .bottom
+
+        return imageView
+    }()
+
+    private let stackView: UIStackView = {
+        let stackView = UIStackView()
+        stackView.translatesAutoresizingMaskIntoConstraints = false
+        stackView.axis = .horizontal
+        stackView.alignment = .center
+        stackView.spacing = Metrics.stackViewSpacing
+        stackView.isUserInteractionEnabled = false
+
+        return stackView
+    }()
+
+    override var intrinsicContentSize: CGSize {
+        return Metrics.contentSize
+    }
+
+    var filterType: AuthorFilterType = .everyone {
+        didSet {
+            switch filterType {
+            case .everyone:
+                authorImageView.image = Gridicon.iconOfType(.multipleUsers, withSize: Metrics.multipleUsersGravatarSize)
+                authorImageView.contentMode = .center
+            case .user(let email):
+                authorImageView.contentMode = .scaleAspectFill
+                if let email = email {
+                    authorImageView.downloadGravatarWithEmail(email, placeholderImage: gravatarPlaceholder)
+                } else {
+                    authorImageView.image = gravatarPlaceholder
+                }
+            }
+        }
+    }
+
+    init() {
+        super.init(frame: .zero)
+
+        commonInit()
+    }
+
+    required init?(coder aDecoder: NSCoder) {
+        super.init(coder: aDecoder)
+
+        commonInit()
+    }
+
+    private func commonInit() {
+        stackView.addArrangedSubview(authorImageView)
+        stackView.addArrangedSubview(chevronImageView)
+        addSubview(stackView)
+        NSLayoutConstraint.activate([
+            stackView.leadingAnchor.constraint(equalTo: leadingAnchor, constant: Metrics.leadingPadding),
+            stackView.centerYAnchor.constraint(equalTo: centerYAnchor),
+            authorImageView.widthAnchor.constraint(equalToConstant: Metrics.gravatarSize.width),
+            authorImageView.heightAnchor.constraint(equalToConstant: Metrics.gravatarSize.height),
+            widthAnchor.constraint(equalToConstant: intrinsicContentSize.width),
+            chevronImageView.heightAnchor.constraint(equalToConstant: Metrics.chevronSize.height + Metrics.chevronVerticalPadding)
+            ])
+
+        authorImageView.image = gravatarPlaceholder
+    }
+
+    private let gravatarPlaceholder: UIImage = Gridicon.iconOfType(.user, withSize: Metrics.gravatarSize)
+
+    private enum Metrics {
+        static let chevronSize = CGSize(width: 10.0, height: 5.0)
+        static let contentSize = CGSize(width: 72.0, height: 44.0)
+        static let gravatarSize = CGSize(width: 28.0, height: 28.0)
+        static let multipleUsersGravatarSize = CGSize(width: 20.0, height: 20.0)
+        static let stackViewSpacing: CGFloat = 7.0
+        static let chevronVerticalPadding: CGFloat = 2.0
+        static let leadingPadding: CGFloat = 12.0
+    }
+}
+
+fileprivate extension UIImage {
```

Comment:
Note on this one! `fileprivate` (AFAIK) in this case behaves exactly as if it was `private`. Accessible by everyone!

Reply:
Whoops, not sure why I did that! Thanks.